### PR TITLE
HIVE-23863 : UGI doAs privilege action  to make calls to Ranger Service

### PR DESCRIPTION
**Pull request by aasha**
## NOTICE

Please create an issue in ASF JIRA before opening a pull request,
and you need to set the title of the pull request which starts with
the corresponding JIRA issue number. (e.g. HIVE-XXXXX: Fix a typo in YYY)
For more details, please see https://cwiki.apache.org/confluence/display/Hive/HowToContribute

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/parse/repl/metric/ReplicationMetricCollector.java [hunk 1]
```diff
@@ -77,6 +77,9 @@ public void reportStageEnd(String stageName, Status status, long lastReplId) thr
       Stage stage = progress.getStageByName(stageName);
       stage.setStatus(status);
       stage.setEndTime(System.currentTimeMillis());
+      if (Status.FAILED == status) {
```
Comment (pkumarsinha) [inline ql/src/java/org/apache/hadoop/hive/ql/parse/repl/metric/ReplicationMetricCollector.java:80]: Could you please give background of this?

File: ql/src/java/org/apache/hadoop/hive/ql/exec/repl/AtlasDumpTask.java [hunk 2]
```diff
@@ -79,6 +80,7 @@ public AtlasDumpTask() {
   @Override
   public int execute() {
     try {
+      SecurityUtils.reloginExpiringKeytabUser();
```
Comment (pkumarsinha) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/repl/AtlasDumpTask.java:83]: Should it be more closure to the actual usage. Meaning, if we are doing a retry inside the task(Repl/Ranger/Atlas) somewhere, should it be  closer to where we are actually interacting with external system?

File: ql/src/java/org/apache/hadoop/hive/ql/parse/repl/metric/ReplicationMetricCollector.java (hunk 1 above)
Comment (aasha) [reply ql/src/java/org/apache/hadoop/hive/ql/parse/repl/metric/ReplicationMetricCollector.java:80]: If the stage has failed, the overall state should also be failed. Now it remains in progress. Noticed during testing this issue.

File: ql/src/java/org/apache/hadoop/hive/ql/exec/repl/AtlasDumpTask.java (hunk 2 above)
Comment (aasha) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/repl/AtlasDumpTask.java:83]: Yes its there as part of the retry as well. This is to make sure any file system calls happening before that is taken care of.

Comment (pkumarsinha) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/repl/AtlasDumpTask.java:83]: Should we have it for Atlas retries as well?

File: ql/src/java/org/apache/hadoop/hive/ql/parse/repl/metric/ReplicationMetricCollector.java (hunk 1 above)
Comment (pkumarsinha) [reply ql/src/java/org/apache/hadoop/hive/ql/parse/repl/metric/ReplicationMetricCollector.java:80]: Sure. Can we have a test for this?

File: ql/src/java/org/apache/hadoop/hive/ql/exec/repl/AtlasDumpTask.java (hunk 2 above)
Comment (aasha) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/repl/AtlasDumpTask.java:83]: Added

File: ql/src/java/org/apache/hadoop/hive/ql/parse/repl/metric/ReplicationMetricCollector.java (hunk 1 above)
Comment (aasha) [reply ql/src/java/org/apache/hadoop/hive/ql/parse/repl/metric/ReplicationMetricCollector.java:80]: Added

**Pull request closed without merging**